packages/mobx-state-tree/src/types/complex-types/model.ts

```diff
@@ -147,17 +148,20 @@ export class ModelType<S, T> extends ComplexType<S, T> implements IModelType<S,
     }
 
     actions<A extends { [name: string]: Function }>(fn: (self: T) => A): IModelType<S, T & A> {
-        const actionInitializer = (self: T) => {
-            this.instantiateActions(self, fn(self))
+        const actionInitializer = (self: T, node: ObjectNode) => {
+            this.instantiateActions(self, fn(self), node)
             return self
         }
         return this.cloneAndEnhance({ initializers: [actionInitializer] })
     }
 
-    instantiateActions(self: T, actions: { [name: string]: Function }) {
+    instantiateActions(self: T, actions: { [name: string]: Function }, node: ObjectNode) {
         // check if return is correct
         if (!isPlainObject(actions))
             fail(`actions initializer should return a plain object containing actions`)
+
+        if (node.type) addReadOnlyProp(node.type, "actions", actions)
```

Comment:
2 questions about this approach

1. potentially several instances could expose different actions (I mean you could return actions conditionally). However I don't think that matters much, as that would be bad practice anyway
2. I think this will break for multiple chained `actions` calls? Or no, probably, it doesn't because each time a new type is produced :thinking: . Let's make sure that is verified by some test.

Reply:
lets' verify both 👍

Reply:
can you make sure that the fields like `actions`  are predeclared on ComplexType? (initialized as `utils.EMPTY_ARRAY`)

Reply:
about 1. i think that's good since the implementation is a representation of the created node, not the type itself.

Reply:
we might want to change the name to indicate that a bit more. e.g. `getMembers(node)`
might be easier to understand and the sugar around maybe too

e.g. 
```js
getMembers(node, publicOnly)
getPrivateMembers(node) // if such ever would be introduced
getActions(node)
...

```